src/Tickets/Commerce/Reports/Data/Order_Summary.php

```diff
@@ -0,0 +1,278 @@
+				'qty_data'      => $quantities,
+				'qty_by_status' => implode( ' | ', array_map( fn( $k, $v ) => "$v $k", array_keys( $quantities ), $quantities ) ),
+			];
+
+			$this->tickets_by_type[ $ticket->type ][] = $ticket_data;
+		}
+	}
+
+	/**
+	 * Process the event sales data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,int> $quantity_by_status The quantity by status.
+	 * @param Ticket_Object $ticket The ticket object.
+	 */
+	protected function process_event_sales_data( array $quantity_by_status, Ticket_Object $ticket ): void {
+		foreach ( $quantity_by_status as $status_slug => $quantity ) {
+			if ( ! isset( $this->event_sales_by_status[ $status_slug ] ) ) {
+				$status = tribe( Status_Handler::class )->get_by_slug( $status_slug );
+
+				// This is the first time we've seen this status, so initialize it.
+				$this->event_sales_by_status[ $status_slug ] = [
+					'label'              => $status->get_name(),
+					'qty_sold'           => 0,
+					'total_sales_amount' => 0,
+					'total_sales_price'  => $this->format_price( 0 ),
+				];
+			}
+			$sales_amount = $quantity * $ticket->price;
+			$this->event_sales_by_status[ $status_slug ]['qty_sold']           += $quantity;
+			$this->event_sales_by_status[ $status_slug ]['total_sales_amount'] += $sales_amount;
+			$this->event_sales_by_status[ $status_slug ]['total_sales_price']  = $this->format_price( $this->event_sales_by_status[ $status_slug ]['total_sales_amount'] );
+
+			// process the total ordered data.
+			$this->total_ordered['qty']    += $quantity;
+			$this->total_ordered['amount'] += $sales_amount;
+			$this->total_ordered['price']  = $this->format_price( $this->total_ordered['amount'] );
+
+			// Only completed orders should be counted in the total sales.
+			if ( Completed::SLUG === $status_slug ) {
+				$this->total_sales['qty']    += $quantity;
+				$this->total_sales['amount'] += $sales_amount;
+				$this->total_sales['price']  = $this->format_price( $this->total_sales['amount'] );
+			}
+		}
+	}
+
+	/**
+	 * Build the event sales data.
+	 *
+	 * @since TBD
+	 */
+	protected function build_event_sales_data(): void {
+		$this->event_sales_data = [
+			'by_status'     => $this->event_sales_by_status,
+			'total_sales'   => $this->total_sales,
+			'total_ordered' => $this->total_ordered,
+		];
+	}
+
+	/**
+	 * Get the tickets.
+	 *
+	 * @since TBD
+	 *
+	 * @return Ticket_Object[]
+	 */
+	public function get_tickets(): array {
+		$provider      = Tribe__Tickets__Tickets::get_event_ticket_provider_object( $this->post_id );
+		$this->tickets = $provider->get_tickets( $this->post_id );
+
+		/**
+		 * Filters the tickets in the order summary report.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $tickets The tickets.
+		 * @param Order_Summary $this The order summary object.
+		 */
+		return apply_filters( 'tec_tickets_commerce_order_report_summary_tickets', $this->tickets, $this );
+	}
+
+	/**
+	 * Get the tickets by type.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
```

Comment:
```suggestion
	 * @return array<string,Ticket[]> A map from ticket types to the Tickets of that type.
```

The `Ticket[]` part might be different, update if required.